src/core/control/pagetype/PageTypeMenu.cpp

```diff
@@ -37,18 +38,20 @@ void PageTypeMenu::loadDefaultPage() {
     setSelected(model.getPageInsertType());
 }
 
-auto PageTypeMenu::createPreviewImage(MainBackgroundPainter* bgPainter, const PageType& pt) -> cairo_surface_t* {
+auto PageTypeMenu::createPreviewImage(const PageType& pt) -> cairo_surface_t* {
     int previewWidth = 100;
     int previewHeight = 141;
     double zoom = 0.5;
 
     auto page = std::make_shared<XojPage>(previewWidth / zoom, previewHeight / zoom);
+    page->setBackgroundType(pt);
 
     cairo_surface_t* surface = cairo_image_surface_create(CAIRO_FORMAT_ARGB32, previewWidth, previewHeight);
     cairo_t* cr = cairo_create(surface);
     cairo_scale(cr, zoom, zoom);
 
-    bgPainter->paint(pt, cr, std::move(page));
+    auto bgView = xoj::view::BackgroundView::createForPage(std::move(page), pt.config, 2.0);
```

Comment:
This seems like a hack: we're creating a new page so that we can draw with the desired width, height, background color, and background type. Could this be refactored?

Reply:
Done!